src/generator.cpp

```diff
@@ -1611,6 +1612,18 @@ std::string Generator::GeneratorImpl::generateVariableInfoObjectString()
                    "<UNITS_SIZE>", std::to_string(unitsSize));
 }
 
+std::string Generator::GeneratorImpl::generateVoiInfoString()
+{
+    std::string component = (mVoi != nullptr)?mVoi->parentComponent()->name():"";
+    std::string name = (mVoi != nullptr)?mVoi->name():"";
+    std::string units = (mVoi != nullptr)?mVoi->units():"";
+
+    return replace(replace(replace(mProfile->voiInfoString(),
+                                   "<COMPONENT>", component),
+                           "<NAME>", name),
+                   "<UNITS>", units);
```

Comment:
If we were handing out awards for super ugly code this would be my nomination.

Reply:
Yeah! :)